lib/cc/analyzer/issue_validator.rb

```diff
@@ -0,0 +1,49 @@
+module CC
+  module Analyzer
+    class IssueValidator
+      CHECKS = [
+        IssueCategoryValidation,
+        IssueCheckNamePresenceValidation,
+        IssueDescriptionPresenceValidation,
+        IssuePathExistenceValidation,
+        IssuePathPresenceValidation,
+        IssueRelativePathValidation,
+        IssueTypeValidation,
+      ].freeze
+
+      attr_reader :error
+
+      def initialize(issue)
+        @issue = issue
+      end
+
+      def valid?
+        @error.blank?
+      end
+
+      def validate
+        if issue && invalid_messages.any?
+          @error = {
+            message: "#{invalid_messages.join("; ")}: `#{issue}`.",
+            issue: issue,
+          }
+          false
+        else
+          true
```

Comment:
From an API design perspective, I guess I also question whether considering a `nil` issue "valid" is semantically a good idea. Personally, I don't think it is.